rpcs3/rpcs3qt/gui_application.cpp

```diff
@@ -354,6 +354,11 @@ void gui_application::InitializeCallbacks()
 		return m_gui_settings->GetBootConfirmation(m_main_window, gui::ib_confirm_fw);
 	};
 
+	callbacks.on_install_fw = [this]()
+	{
+		if (m_main_window) m_main_window->InstallPup();
```

Comment:
There are some real problems here.
- Cancelling the file explorer will simply boot the game instead of handling things gracefully. This looks like a bug from user perspective.
- InstallPup is just semi-blocking. This means you get to choose a PUP file, then the game continues, then a dialog might pop up that asks if you wanna install v.4.85 or whatever, and finally the just-booted game stops and the firmware is installed.
- After installing the firmware, the game doesn't continue.

I'm starting to wonder if you shouldn't just add a debug setting for booting without firmware instead of allowing users to boot without. it and adding a clunky dialog that complicates things for no reason.

Reply:
Fixed.